### PR TITLE
[Perf] Avoid pageable HtoD transfer in MinTokensLogitsProcessor

### PR DESCRIPTION
**Pull request by jthomson04**
## Purpose
The `MinTokensLogitsProcessor` introduces a pageable HtoD transfer here: https://github.com/vllm-project/vllm/blob/342c4f147266e7d01a8a6a64754df7e0f7f8991a/vllm/v1/sample/logits_processor/builtin.py#L232. This is problematic, since it is synchronizing wrt the device. When running with `--async-scheduling`, this synchronizing behavior causes the scheduler to no longer be overlapped with model execution. At low concurrencies, this leads to a significant (up to 30%) increase in TPOT when including min_tokens in the request.

I collected some nsys traces illustrating this.

Without min_tokens in request:
image

With min_tokens in request:
image

## Test Plan
Existing tests for `MinTokensLogitsProcessor` are sufficient; no further tests are required.

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/v1/sample/logits_processor/builtin.py [hunk 1]
```diff
@@ -178,6 +178,13 @@ def __init__(
             self._device_tensor([], torch.int32),
         )
 
+        self.neg_inf_tensor = torch.full(
+            (),
+            -float("inf"),
+            dtype=torch.float32,
+            device=device,
+        )
```
Comment (njhill) [inline vllm/v1/sample/logits_processor/builtin.py]: Just use `torch.tensor(-float("inf"), ...`?

File: vllm/v1/sample/logits_processor/builtin.py [hunk 2]
```diff
@@ -110,7 +110,7 @@ def apply(self, logits: torch.Tensor) -> torch.Tensor:
         # Identify valid tokens using threshold comparison
         invalid_token_mask = probability_values < adjusted_min_p
         # Apply mask using boolean indexing
-        logits[invalid_token_mask] = -float("inf")
+        logits[invalid_token_mask] = self.neg_inf_tensor
```
[the remark this answers is not in the document]
Comment (njhill) [reply vllm/v1/sample/logits_processor/builtin.py]: This appears to be correct - you need to create this in the init method of this class too

**Review by njhill — COMMENTED** [review]
Great catch, thanks @jthomson04!

File: vllm/v1/sample/logits_processor/builtin.py (hunk 2 above)
Comment (nvpohanh) [inline vllm/v1/sample/logits_processor/builtin.py]: could we try `logits[invalid_token_mask].fill_(-float("inf"))` so that we can avoid creating `neg_inf_tensor`?

**Review by nvpohanh — APPROVED** [review]

File: vllm/v1/sample/logits_processor/builtin.py (hunk 1 above)
Comment (nvpohanh) [reply vllm/v1/sample/logits_processor/builtin.py]: I asked @jthomson04 to remove `neg_inf_tensor`

**Review by njhill — APPROVED** [review]

**Comment by njhill** [comment]
The test failure looks likely related given it's to a logitsproc test, I can look into it soon.
